tests/test_multiple_nodes.py

```diff
@@ -51,4 +52,4 @@ async def test_many_servers_many_clients(num_servers, num_clients):
         clients = []
         for __ in range(i, min(i + somaxconn, num_clients * num_servers)):
             clients.append(client_node(listeners[__ % num_servers].port))
-        await asyncio.gather(*clients, loop=asyncio.get_event_loop())
+        await asyncio.gather(*clients, loop=get_event_loop())
```

Comment:
Python 3.10 doesn't have the loop keyword argument to `asyncio.gather` (and it is deprecated since 3.8, so probably just remove this keyword argument). The relevant event loop is grabbed from the coroutines/futures being passed in.

For coroutines, the currently running event loop is used to run the coroutines. For futures, the loop that created the future is used to run things (but all futures must come from the same loop).

So perhaps:

```
from itertools import cycle, islice

...
    somaxconn = get_somaxconn()

    listeners = cycle((ucp.create_listener(server_node) for _ in range(num_servers)))

    # We ensure no more than `somaxconn` connections are submitted
    # at once. Doing otherwise can block and hang indefinitely.
    for _ in range(0, num_clients * num_servers, somaxconn):
        await asyncio.gather(
            *(client_node(listener.port) for listener in islice(listeners, somaxconn))
        )
```

Reply:
(This is the only call to `asyncio.gather` that uses the loop keyword argument in the codebase, so no need to fix anywhere else)